src/Azure.Functions.Cli/Actions/LocalActions/CreateFunctionAction.cs

```diff
@@ -19,18 +25,100 @@ internal class CreateFunctionAction : BaseAction
 
-        public CreateFunctionAction(ITemplatesManager templatesManager, ISecretsManager secretsManager)
+        public CreateFunctionAction(ITemplatesManager templatesManager, ISecretsManager secretsManager, ISettings settings)
         {
             _templatesManager = templatesManager;
             _secretsManager = secretsManager;
+            _settings = settings;
         }
 
+        private async Task<(bool, string)> CallAzCommandAsync(string commandName)
+        {
+            var windowscmd = new StringBuilder();
+            windowscmd.Append("/c az ").Append(commandName);
+
+            var az = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                        ? new Executable("cmd", windowscmd.ToString())
+                        : new Executable("az", commandName);
+
+            var stdout = new StringBuilder();
+            var stderr = new StringBuilder();
+            var exitCode = await az.RunAsync(o => stdout.AppendLine(o), e => stderr.AppendLine(e));
+            if (exitCode == 0)
+            {
+                return (true, (stdout.ToString().Trim(' ', '\n', '\r', '"')));
+            }
+            else
+            {
+                return (false, (stderr.ToString().Trim(' ', '\n', '\r'))); // "Unable to connect to Azure. Make sure you have the `az` CLI installed and logged in and try again";
+            }
+        }
+
+        private async Task CreateWebJobsStorageKeyAsync()
+        {
+            ColoredConsole.Write("Please enter the name of your resource group for your storage account: ");
+            string resource_group = Console.ReadLine();
+            
+            ColoredConsole.Write("Please enter the name of your storage account: ");
+            string storage_account = Console.ReadLine();
+
+            //verify if the resource group and account exists and get the details
+            //az storage account show -g MyResourceGroup -n MyStorageAccount
+            var command = new StringBuilder();
+            command.Append("storage account keys list -g ").Append(resource_group).Append(" -n ").Append(storage_account).Append(" --output json");
+            (bool success, string keys) = await CallAzCommandAsync(command.ToString());
+
+            if (success)
+            {
+                JArray json_keys = JArray.Parse(keys);
+                string account_key = json_keys[0].Value<string>("value");
+                //"{ACCOUNT_NAME}_STORAGE": "DefaultEndpointsProtocol=https;AccountName={ACCOUNT_NAME};AccountKey={ACCOUNT_KEY}",
+                var storage_value = new StringBuilder();
+                storage_value.Append("DefaultEndpointsProtocol=https;AccountName=").Append(storage_account).Append(";AccountKey=").Append(account_key);
+                _secretsManager.SetSecret("AzureWebJobsStorage", storage_value.ToString());
+            }
+            else
+            {
+                ColoredConsole.WriteLine(VerboseColor($"Unable to fetch key token from az cli. Error: {keys}"));
```

Comment:
You might want to `throw new CliException(<message>);` here to make sure we don't continue running if we hit this state.